frame/pools/src/lib.rs

```diff
@@ -0,0 +1,1135 @@
+				delegator: who,
+				pool: target,
+				bonded: exact_amount_to_bond,
+			});
+
+			Ok(())
+		}
+
+		/// A bonded delegator can use this to claim their payout based on the rewards that the pool
+		/// has accumulated since their last claimed payout (OR since joining if this is there first
+		/// time claiming rewards).
+		///
+		/// Note that the payout will go to the delegator's account.
+		#[pallet::weight(666)]
+		pub fn claim_payout(origin: OriginFor<T>) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let delegator = Delegators::<T>::get(&who).ok_or(Error::<T>::DelegatorNotFound)?;
+			let bonded_pool = BondedPools::<T>::get(&delegator.pool).ok_or_else(|| {
+				log!(error, "A bonded pool could not be found, this is a system logic error.");
+				Error::<T>::PoolNotFound
+			})?;
+
+			Self::do_reward_payout(who, delegator, &bonded_pool)?;
+
+			Ok(())
+		}
+
+		/// A bonded delegator can use this to unbond _all_ funds from the pool.
+		/// In order to withdraw the funds, the delegator must wait
+		#[pallet::weight(666)]
+		pub fn unbond(origin: OriginFor<T>, num_slashing_spans: u32) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+			let delegator = Delegators::<T>::get(&who).ok_or(Error::<T>::DelegatorNotFound)?;
+			let mut bonded_pool =
+				BondedPools::<T>::get(&delegator.pool).ok_or(Error::<T>::PoolNotFound)?;
+
+			// Claim the the payout prior to unbonding. Once the user is unbonding their points
+			// no longer exist in the bonded pool and thus they can no longer claim their payouts.
+			// It is not strictly necessary to claim the rewards, but we do it here for UX.
+			Self::do_reward_payout(who.clone(), delegator, &bonded_pool)?;
+
+			// Re-fetch the delegator because they where updated by `do_reward_payout`.
+			let mut delegator = Delegators::<T>::get(&who).ok_or(Error::<T>::DelegatorNotFound)?;
+			// Note that we lazily create the unbonding pools here if they don't already exist
+			let sub_pools = SubPoolsStorage::<T>::get(&delegator.pool).unwrap_or_default();
+			let current_era = T::StakingInterface::current_era().unwrap_or(Zero::zero());
+
+			let balance_to_unbond =
+				bonded_pool.balance_to_unbond(&delegator.pool, delegator.points);
+
+			// Update the bonded pool. Note that we must do this *after* calculating the balance
+			// to unbond so we have the correct points for the balance:share ratio.
+			bonded_pool.points = bonded_pool.points.saturating_sub(delegator.points);
+
+			// Call withdraw unbonded to minimize unlocking chunks. If this is not done then we
+			// would have to rely on delegators calling `withdraw_unbonded` in order to clear
+			// unlocking chunks. This is a catch 22 for delegators who have not yet unbonded
+			// because the pool needs to call `withdraw_unbonded` so they can `unbond`, but they
+			// must call `unbond` prior to being able to call `withdraw_unbonded`.
+			T::StakingInterface::withdraw_unbonded(delegator.pool.clone(), num_slashing_spans)?;
```

Comment:
We can make a permissionless withdraw unbonded extrinsic that simply calls withdraw unbonded for the underlying pool and then we can remove this call here